### PR TITLE
Refactor invite and membership specs

### PR DESCRIPTION
**Pull request by tommotaylor**
Improves two things:
- Naming of spec files so that they are easier to understand
- Removes some duplication we had

### CODE REVIEW

File: spec/features/memberships/invite_user_to_second_organisation_spec.rb [hunk 1]
```diff
@@ -44,17 +50,21 @@
     end
   end
 
-  context 'when a super admin does not have can_manage_team permissions' do
```
Comment (tommotaylor) [inline spec/features/memberships/invite_user_to_second_organisation_spec.rb]: This was moved to `specs/super_admin/`

File: spec/features/confirm_membership_invitation_spec.rb [hunk 2]
```diff
@@ -1,26 +0,0 @@
-describe 'Confirming an invitation to a new organisation', type: :feature do
-  let(:organisation) { create(:organisation) }
```
Comment (tommotaylor) [inline spec/features/confirm_membership_invitation_spec.rb]: Duplicate of `'Confirming an invite to a second or subsequent organisation'` below

File: spec/features/invite_existing_user_spec.rb [hunk 3]
```diff
@@ -1,65 +0,0 @@
-require 'support/invite_use_case'
```
Comment (tommotaylor) [inline spec/features/invite_existing_user_spec.rb]: Duplicate file

**Comment by emileswarts** [comment]
Nice, looks good

**Pull request merged**